crates/viewer/re_viewer/src/app.rs

```diff
@@ -2275,6 +2406,163 @@ impl App {
         }
     }
 
+    #[cfg(not(target_arch = "wasm32"))]
+    fn try_enqueue_arrow_ingestion(
+        &mut self,
+        store_id: &StoreId,
+        arrow_msg: &re_log_types::ArrowMsg,
+        channel_source: Arc<SmartChannelSource>,
+        msg_will_add_new_store: bool,
+    ) -> bool {
+        let arrow_msg = arrow_msg.clone();
+
+        let promise = poll_promise::Promise::spawn_thread("prepare_arrow", move || {
+            PreparedArrowChunk::from_arrow_msg(&arrow_msg)
+        });
```

Comment:
thread spawning is very expensive and for every single message you spawn a new one where the only thing it does is to unpack the chunk from the message?